File: generatePortfolioPDF.js
const puppeteer = require('puppeteer');
const path = require('path');
const fs = require('fs');

async function generatePortfolioPDF() {
  try {
    // Check if index.html exists
    const htmlPath = path.join(__dirname, 'index.html');
    if (!fs.existsSync(htmlPath)) {
      console.log(`Error: HTML file not found at ${htmlPath}`);
      return;
    }

    // Launch the browser
    const browser = await puppeteer.launch();
    const page = await browser.newPage();

    // Load the HTML file
    const fileUrl = `file://${htmlPath}`;
    await page.goto(fileUrl, {
      waitUntil: 'networkidle0'
    });

    // Generate PDF
    const pdfPath = path.join(__dirname, 'portfolio.pdf');
    await page.pdf({
      path: pdfPath,
      format: 'A4',
      printBackground: true,
      margin: {
        top: '20px',
        bottom: '20px',
        left: '20px',
        right: '20px'
      }
    });

    await browser.close();

    // Get file size
    const stats = fs.statSync(pdfPath);
    const fileSizeKB = stats.size / 1024;

    console.log(`Portfolio PDF successfully generated at: ${pdfPath}`);
    console.log(`File size: ${fileSizeKB.toFixed(2)} KB`);
  } catch (error) {
    console.error('Error generating PDF:', error);
  }
}

generatePortfolioPDF();

File: generate_portfolio_pdf.py
import weasyprint
import os

def generate_portfolio_pdf():
    # Get the current directory
    current_dir = os.path.dirname(os.path.abspath(__file__))
    
    # Path to your index.html file
    html_file = os.path.join(current_dir, 'index.html')
    
    # Output PDF file path
    pdf_file = os.path.join(current_dir, 'portfolio.pdf')
    
    # Check if HTML file exists
    if not os.path.exists(html_file):
        print(f"Error: HTML file not found at {html_file}")
        return
    
    # Generate PDF
    try:
        # Read the HTML file
        with open(html_file, 'r', encoding='utf-8') as f:
            html_content = f.read()
        
        # Convert HTML to PDF
        html_doc = weasyprint.HTML(string=html_content, base_url=current_dir)
        html_doc.write_pdf(pdf_file)
        
        print(f"Portfolio PDF successfully generated at: {pdf_file}")
        print(f"File size: {os.path.getsize(pdf_file) / 1024:.2f} KB")
        
    except Exception as e:
        print(f"Error generating PDF: {str(e)}")

if __name__ == "__main__":
    generate_portfolio_pdf()